CMakeLists.txt

```diff
@@ -1,54 +1,102 @@
 cmake_minimum_required(VERSION 3.5)
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_C_STANDARD 99)
+set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+set(CMAKE_POLICY_DEFAULT_CMP0063 NEW)
 project(Fossilize LANGUAGES CXX C)
 
-option(FOSSILIZE_MSVC_ANALYZE "Enable /analyze" OFF)
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+set(CMAKE_THREA_PREFER_PTHREAD ON)
+
+include(CMakeDependentOption)
+include(GNUInstallDirs)
+include(CTest)
+
+find_package(Threads REQUIRED)
+find_package(Vulkan)
```

Comment:
External Vulkan is not needed.

Reply:
I forgot to remove this, thanks for the catch. Fossilize doesn't work with stock Vulkan anyhow.

Reply:
By stock Vulkan, do you mean trying to link against Vulkan with the package?

Reply:
The one I installed from LunarXChange, yes. Headers are not included as `<vulkan/vulkan.h>`, but rather as `"vulkan.h"`. There are also several headers missing from 'stock' Vulkan.

When you "link" against a library in CMake, it'll perform something known as transitive linking, where additional properties, such as `[INTERFACE_]INCLUDE_DIRECTORIES`, `COMPILE_OPTIONS`, etc. are used. The only time this doesn't occur is when the `TYPE`  of a target is `OBJECT`, where you need to do `$<TARGET_OBJECTS:${my-target}>` but this was fixed in a more recent version of CMake, though it requires setting a policy, and if your minimum is older it obviously won't have the intended effect. Even if I were to just do
`target_include_directories(fossilize PRIVATE $<TARGET_PROPERTY:Vulkan::Vulkan,INCLUDE_DIRECTORIES>)`, fossilize will fail to compile.

Reply:
This has been removed as of the latest commit.